### PR TITLE
refactor: Only output NPM logs if there was an error

### PR DESCRIPTION
**Pull request by jshearer**
**Description:**

This is a pragmatic first step towards https://github.com/estuary/animated-carnival/issues/64. Ultimately we'll have structured logging and can filter logs after-the-fact, but in the mean time this does a lot to clean up the log output.

_Before: _
Screen Shot 2022-09-12 at 17 05 36

_After:_
Screen Shot 2022-09-12 at 17 03 37

FWIW there _is_ still a lot of extraneous log output in here (do I need to be concerned about the warning telling me about slow etcd, if the result is still a success? I'd argue no), but judging by the size of the scrollbars, NPM was the vast majority of the noise.

**Testing:**

I wasn't able to figure out how to cause an error, so there's a good chance this'll blow up in the case of an error. @jgraettinger any suggestions for causing an NPM error?

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/665)

### CODE REVIEW

**Comment by mdibaiee** [comment]
@jshearer to cause an error you can try creating a salesforce capture, since that one fails because of #641

**Comment by jshearer** [comment]
Ok so I added the `docker pull --quiet` flag here, and also tested with a failing derivation, and it appears to work.

Output with typo'd derivation:
Screen Shot 2022-09-20 at 09 32 56

Output with fixed derivation:
Screen Shot 2022-09-20 at 09 34 53

File: crates/build/src/lib.rs [hunk 1]
```diff
@@ -237,14 +237,17 @@ fn npm_cmd(package_dir: &std::path::Path, args: &[&str]) -> Result<(), anyhow::E
...
-
-    if !status.success() {
-        anyhow::bail!("npm command {:?} failed with {}", args.join(" "), status);
+    let result = cmd
+        .output()
+        .context("failed to spawn `npm` command");
+
+    match result {
+        Ok(output) if !output.status.success() => {
+            tracing::error!(npm_command = args.join(" ").as_str(), "{:?}", &String::from_utf8(output.stdout));
```
[the remark this answers is not in the document]
Comment (jshearer) [reply crates/build/src/lib.rs]: Done and done

**Review by jgraettinger — APPROVED** [review]
LGTM

> Reviewed 1 of 1 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/estuary/flow/665#-:-NCWEG6JCQenKqwGxBvg:bb6ywmx)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @jshearer)

**Pull request merged**